### DISCUSSION TITLE
Registering new app to use localhost address

### DISCUSSION THREAD

**Discussion by mckreeger** [etsy/open-api#1052] [Q&A]
Under the v2 API when registering an app, the website URL was NOT required, and localhost could be used.
Under the v3 API, the website URL is required and localhost is not considered a valid value.

But if you follow the tutorials for v3 API (https://developers.etsy.com/documentation/tutorials/quickstart) the example is a "localhost" based app with a localhost redirect_uri. Fortunately, I already has an app registered under v2 and managed to add callback URLs to that old definition.

However when attempting to create a new app, the app was "rejected" with a "Missing valid website URL", and when I try and at least edit the application I get sent to an error page (https://www.etsy.com/error.php).

And when I updated the callback URLs on the old app, the only ones that appear to have worked ended in "oauth/redirect" which matches essentially the one defined in the "quickstart" example.

The registration process is broken and the "website URL" should not be required, or at least accept "http://localhost" or default to that unless the developer specifies something else. Did anyone at etsy actually walk through their own startup guide? As it stands now, using "http://localhost" is required in the "website URL" in order for it to accept the "Callback URLs" with "http://localhost" but I can't register a new app with "http://localhost".

If you feel that I have recursively repeated the same thing, please start at the top of this discussion and reread. I had to experiment multiple times to get the "right" and answer on the old v2 registered app because there were no clear succinct answers anywhere.

**Comment by thisisjake** [comment]
I also wasn't able to authenticate using localhost with my local application a year or so ago.

I decided to use Postman to get the credentials instead. I haven't needed to re-authenticate since. Just ensure that you make at least one request every 90 days to so that you have a 'fresh' refresh token.

Unfortunately if you need to change the scope or your refresh token expires then you'll need to go through the authentication process again (however it's not like it takes a long time).

If you want Etsy to fix the above, best to email them as they don't monitor here anymore unfortunately.
